### ISSUE TITLE
Clean up use of unittest assert functions in twisted/test

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8391]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8391|
|Type|enhancement|
|Created|2016-06-03 02:15:30Z|
|Branch|https://github.com/twisted/twisted/tree/assertfns-8391|

This ticket is split out from [ticket:8371] because the patch there was too big to review.

The files in twisted/test
should have their use of unittest assert functions cleaned up so that:

* the most appropriate unittest assert function is used appropriately in each file
* use of deprecated aliases should be eliminated, to reduce deprecation
  warnings and errors when moving to Python3 compatibility

This has some pointers about deprecated unittest assert aliases:

See: https://docs.python.org/2.7/library/unittest.html#deprecated-aliases

|= Method Name =|= Deprecated alias(es) =|
|---------------|------------------------|
| assertEqual() | failUnlessEqual, assertEquals |
| assertNotEqual() | failIfEqual |
| assertTrue() | failUnless, assert_ |
| assertFalse() | failIf |
| assertRaises() |	failUnlessRaises |
| assertAlmostEqual() | failUnlessAlmostEqual  |
| assertNotAlmostEqual() | failIfAlmostEqual  |

<details><summary>Searchable metadata</summary>

```
trac-id__8391 8391
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__assertfns_8391 assertfns-8391
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464920130115220 1464920130115220
changetime__1474198049187875 1474198049187875
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/107

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for your contribution, rodrigc.

As with [#8376](https://github.com/twisted/twisted/issues/8376), the changes to these assertion functions are over-broad.  These are a lot of changes to look over, but please re-submit more minimal changes that just address the necessary bits to migrate to python 3.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc removed owner|
|-|-|

OK, I uploaded a new diff which just removes deprecated aliases.
​https://github.com/twisted/twisted/pull/107

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

Getting a little tired for a full review now, so I'll just leave this as a comment, but I ran this through the buildbot and twistedchecker complained: https://buildbot.twistedmatrix.com/builders/twistedchecker/builds/154/steps/run-twistedchecker/logs/new%20twistedchecker%20errors

If you can take care of it before I wake up that would speed up me merging it, but even otherwise I think I'll probably get to it today.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

(by "get to it" meaning "fix the minor quibbles and merge")

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_7"></a>|@wsanchez set owner to @rodrigc|
|-|-|

The quibbles glyph noted are specifically:

```
************* Module twisted.test.test_application
C0301:189,0: Line too long (86/79)
************* Module twisted.test.test_text
C0301:235,0: Line too long (96/79)
************* Module twisted.test.test_unix
C0301:350,0: Line too long (87/79)
```

There are also some cases here of `assertTrue(not thing)` and `assertTrue(isinstance(obj))` which should ideally be redone as `assertFalse` and `assertIsinstance` as Adi noted in a different ticket.  I wouldn't block merge for that, because your change fixes one thing and fixing more isn't a prerequisite for merging, but if you have a moment to clean that up, this would be a nice opportunity to do so.

Back to rodrigc for at least the line too long complaints.  If you at least clean that up, you're good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_8"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset dec29795589d546dcb980a71ea602de2658a942d
```
#!CommitTicketReference repository="" revision="dec29795589d546dcb980a71ea602de2658a942d"
Merge 'assertfns-8391': Clean up use of unittest assert functions

Author: rodrigc
Reviewer: wsanchez, glyph
Fixes: #8391
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_9"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
